newsroom/push/publishing.py

```diff
@@ -101,14 +107,31 @@ async def publish_item(self, doc: Item, original: Item):
 
         signals.publish_item.send(app, item=doc, is_new=original is None)
 
-        _id = service.create([doc])[0]
-        if "associations" not in doc and original is not None and bool(original.get("associations", {})):
-            service.patch(_id, updates={"associations": None})
+        doc_id = await self.publish_doc_to_content_api(doc)
         if "evolvedfrom" in doc and parent_item:
-            service.system_update(parent_item["_id"], {"nextversion": _id}, parent_item)
-        return _id
+            await wire_search.service.system_update(parent_item.id, {"nextversion": doc_id})
+        return doc_id
+
+    async def publish_doc_to_content_api(self, item_dict: dict[str, Any]) -> str:
+        # Copied from content_api.publish.utils.publish_doc_to_content_api
```

Comment:
```suggestion
        # Copied from content_api.publish.utils.publish_doc_to_content_api
```
is there a way to reuse it instead or there is an intention to remove this repeated code?

Reply:
There are subtle changes with this one vs the one in the ContentAPI, such as using the WireItem model instead of ContentAPIItem model (with changes to the fields & schema). Kept it here because it's a small function, and barely ever changed in both Newshub and ContentAPI for many years. Last change in ContentAPI publishing, in regards to Newshub usage, was 4+ years ago.